backend/database/schema.prisma

```diff
@@ -24,6 +24,7 @@ model Collection {
     description             String?
     embedder_config         Json
     // Collection can have multiple data sources
+    // TODO (chiragjn): Why does this have to be Nullable? Default to {}
```

Comment:
While creating collection, user might not provide a datasource via API. Hence initially it can be None

Reply:
The reason being here, I checked, first the collection is created whose pydantic `CreateCollection` does not have associated data sources, then the source is added later in the API.